File: languages/php/Bloock/Entity/Identity/CredentialBody.php
<?php

namespace Bloock\Entity\Identity;

use Google\Protobuf\Internal\RepeatedField;

class CredentialBody
{
    private RepeatedField $context;
    private string $id;
    private RepeatedField $type;
    private string $issuanceDate;
    private $credentialSubject;
    private CredentialStatus $credentialStatus;
    private string $issuer;
    private CredentialSchema $credentialSchema;
    private CredentialProof $credentialProof;

    /**
     * @param RepeatedField $context
     * @param string $id
     * @param RepeatedField $type
     * @param string $issuanceDate
     * @param $credentialSubject
     * @param CredentialStatus $credentialStatus
     * @param string $issuer
     * @param CredentialSchema $credentialSchema
     * @param CredentialProof $credentialProof
     */
    public function __construct(RepeatedField $context, string $id, RepeatedField $type, string $issuanceDate, $credentialSubject, CredentialStatus $credentialStatus, string $issuer, CredentialSchema $credentialSchema, CredentialProof $credentialProof)
    {
        $this->context = $context;
        $this->id = $id;
        $this->type = $type;
        $this->issuanceDate = $issuanceDate;
        $this->credentialSubject = $credentialSubject;
        $this->credentialStatus = $credentialStatus;
        $this->issuer = $issuer;
        $this->credentialSchema = $credentialSchema;
        $this->credentialProof = $credentialProof;
    }

    public static function fromProto(\Bloock\CredentialBody $res): CredentialBody
    {
        return new CredentialBody(
            $res->getContext(),
            $res->getId(),
            $res->getType(),
            $res->getIssuanceDate(),
            $res->getCredentialSubject(),
            CredentialStatus::fromProto($res->getCredentialStatus()),
            $res->getIssuer(),
            CredentialSchema::fromProto($res->getCredentialSchema()),
            CredentialProof::fromProto($res->getProof())
        );
    }

    /**
     * @return RepeatedField
     */
    public function getContext(): RepeatedField
    {
        return $this->context;
    }

    /**
     * @return string
     */
    public function getId(): string
    {
        return $this->id;
    }

    /**
     * @return RepeatedField
     */
    public function getType(): RepeatedField
    {
        return $this->type;
    }

    /**
     * @return string
     */
    public function getIssuanceDate(): string
    {
        return $this->issuanceDate;
    }

    /**
     * @return mixed
     */
    public function getCredentialSubject()
    {
        return $this->credentialSubject;
    }

    /**
     * @return CredentialStatus
     */
    public function getCredentialStatus(): CredentialStatus
    {
        return $this->credentialStatus;
    }

    /**
     * @return string
     */
    public function getIssuer(): string
    {
        return $this->issuer;
    }

    /**
     * @return CredentialSchema
     */
    public function getCredentialSchema(): CredentialSchema
    {
        return $this->credentialSchema;
    }

    public function toProto(): \Bloock\CredentialBody
    {
        $p = new \Bloock\CredentialBody();
        $p->setContext($this->context);
        $p->setId($this->id);
        $p->setType($this->type);
        $p->setIssuanceDate($this->issuanceDate);
        $p->setCredentialSubject($this->credentialSubject);
        $p->setCredentialStatus($this->credentialStatus->toProto());
        $p->setIssuer($this->issuer);
        $p->setCredentialSchema($this->credentialSchema->toProto());
        $p->setProof($this->credentialProof->toProto());
        return $p;
    }

    /**
     * @return CredentialProof
     */
    public function getCredentialProof(): CredentialProof
    {
        return $this->credentialProof;
    }
}

File: languages/python/bloock/entity/identity/credential_body.py
from __future__ import annotations

from typing import List

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.credential_proof import CredentialProof
from bloock.entity.identity.credential_schema import CredentialSchema
from bloock.entity.identity.credential_status import CredentialStatus


class CredentialBody:
    def __init__(self, context: List[str], id: str, type: List[str], issuance_date: str, credential_subject: str,
                 credential_status: CredentialStatus, issuer: str, credential_schema: CredentialSchema,
                 proof: CredentialProof) -> None:
        self.context = context
        self.id = id
        self.type = type
        self.issuance_date = issuance_date
        self.credential_subject = credential_subject
        self.credential_status = credential_status
        self.issuer = issuer
        self.credential_schema = credential_schema
        self.proof = proof

    @staticmethod
    def from_proto(c: proto.CredentialBody) -> CredentialBody:
        return CredentialBody(
            context=c.context,
            id=c.id,
            type=c.type,
            issuance_date=c.issuance_date,
            credential_subject=c.credential_subject,
            credential_status=CredentialStatus.from_proto(c.credential_status),
            issuer=c.issuer,
            credential_schema=CredentialSchema.from_proto(c.credential_schema),
            proof=CredentialProof.from_proto(c.proof)
        )

    def to_proto(self) -> proto.CredentialBody:
        return proto.CredentialBody(
            context=self.context,
            id=self.id,
            type=self.type,
            issuance_date=self.issuance_date,
            credential_subject=self.credential_subject,
            credential_status=self.credential_status.to_proto(),
            issuer=self.issuer,
            credential_schema=self.credential_schema.to_proto(),
            proof=self.proof.to_proto()
        )
